pandas/core/categorical.py

```diff
@@ -263,18 +259,12 @@ def __init__(self, values, categories=None, ordered=None, name=None, fastpath=Fa
 
         if categories is None:
             try:
-                codes, categories = factorize(values, sort=True)
-                # If the underlying data structure was sortable, and the user doesn't want to
-                # "forget" this order, the categorical also is sorted/ordered
-                if ordered is None:
-                    ordered = True
+                codes, categories = factorize(values, sort=ordered)
             except TypeError:
-                codes, categories = factorize(values, sort=False)
-                if ordered:
-                    # raise, as we don't have a sortable data structure and so the user should
-                    # give us one by specifying categories
-                    raise TypeError("'values' is not ordered, please explicitly specify the "
-                                    "categories order by passing in a categories argument.")
+                # raise, as we don't have a sortable data structure and so the user should
```

Comment:
same here, just replace the `if ordered:` part with a warning that categories were not lexi-sorted and the user has to specify one, if he wants to have the categories in a specific order (!= the Categorical ordered)